### PR TITLE
[FEATURE] add npm_script Task 

### PR DESCRIPTION
**Pull request by Kanti**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | #208

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

**Comment by Kanti** [comment]
Please Review! 😅

**Comment by veewee** [comment]
Thanks @Kanti! The code looks good.

2 small questions:

- Maybe it's better to set the defaults for the `triggered_by` option since it mostly is triggered by js / css changes? These are the defaults for Grunt: [js, jsx, coffee, ts, less, sass, scss]
- Is it possible to specify the location of the `package.json` file for people with an exotic file structure? Can't find how to do it in the documentation, but it could also be done by specifying a working directory on the process.

**Comment by Kanti** [comment]
added changes Please Review 😅

File: src/GrumPHP/Task/NpmScript.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        $files = $context->getFiles()->extensions($config['triggered_by']);
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('npm');
+        $arguments->addRequiredArgument('%s', $config['script']);
+
+        $process = $this->processBuilder->buildProcess($arguments);
+        $process->setWorkingDirectory($config['working_directory']);
```
Comment (veewee) [inline src/GrumPHP/Task/NpmScript.php]: Will this use the correct directory if you add relative directories? Maybe we should make it relative to the getcwd() ?

**Comment by veewee** [comment]
Thanks!

**Pull request merged**